spec/rubocop/cop/rspec/predicate_matcher_spec.rb

```diff
@@ -91,6 +91,8 @@
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `be_something` matcher over `something?`.
           expect(foo.has_key?('foo')).to be_truthy
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `have_key` matcher over `has_key?`.
+          expect(foo.key?('foo')).to be_truthy
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `have_key` matcher over `key?`.
```

Comment:
Does this imply that the receiver is an instance of a Hash?
If it’s e.g. an SqlColumn, have_key? won’t fit, and it should be is_key? instead

Reply:
Can we make a distinction at least if there’s an argument?

Reply:
TIL key? is aliased as has_key? https://ruby-doc.org/3.4.1/Hash.html#method-i-include-3F

Reply:
Proposal:
Only auto-correct expect(foo.key?(foo)) with a single argument.
Only raise an offence with zero or multiple args or a block

Reply:
To be honest, I don't really like this cop (or others) that make broad assumptions about types which can lot safely be deduced with static analysis. There will _always_ be false positives.

Reply:
> TIL key? is aliased as has_key?

And I thought one of them had been deprecated, but apparently that's not the case.

I also think this method is aliased as `member?` and `includes?` btw.